src/SGFParser.cpp

```diff
@@ -38,6 +38,81 @@
 #include "SGFTree.h"
 #include "Utils.h"
 
+std::vector<SGFParser::ELF_Data> SGFParser::chop_elf(std::string filename,
+                                                     size_t stopat) {
+    std::ifstream ins(filename.c_str(), std::ifstream::binary | std::ifstream::in);
+
+    if (ins.fail()) {
+        throw std::runtime_error("Error opening file");
+    }
+
+    std::vector<ELF_Data> result;
+    std::string gamebuff;
+
+    auto constexpr MAX = std::numeric_limits<std::streamsize>::max();
+    std::string str;
+    char c;
+    int num_move;
+    int total_policy;
+
+    while (result.size() <= stopat) {
+        while (true) {
+            ins.ignore(MAX, '\"');
+            if ((ins.rdstate() & std::ifstream::eofbit) != 0) return result;
+            std::getline(ins, str, ':');
+            if (str == "content\"") break;
+        }
+        result.emplace_back(ELF_Data());
+        auto& data = result.back();
```

Comment:
Don't need this trick with a vector, they're almost free to move.

Reply:
ELF_Data is a struct consisting of a std::string, a float, and a std::vector, so I think this is better. Any argument against this "trick"?